sbt

```diff
@@ -278,9 +280,28 @@ acquire_sbt_jar () {
   } || {
     sbt_jar="$(jar_file "$sbt_version")"
     download_url "$(make_url "$sbt_version")" "$sbt_jar"
+
+    case "${sbt_version}" in
+      0.*) vlog "SBT versions < 1.0 do not have published MD5 checksums, skipping check"; echo "" ;;
+        *) verify_sbt_jar "${sbt_jar}" ;;
+    esac
   }
 }
 
+verify_sbt_jar() {
+  local jar="${1}"
+  local md5="${jar}.md5"
+
+  download_url "$(make_url "${sbt_version}").md5" "${md5}" true > /dev/null 2>&1
+  if echo "$(cat "${md5}")  ${jar}" | md5sum -c -; then
+    rm -f "${md5}"
+    return 0
+  else
+    echoerr "Checksum does not match"
+    return 1
```

Comment:
If I follow the logic properly this will result in "Checksum does not match" and then "Download failed. Obtain the jar manually and place it at $sbt_jar".  If checksums don't match is it fair to say "Download fail"?  I'm genuinely not sure.  WDYT?

Reply:
Maybe re-phrasing the message may be better.
We could not download *and validate* the file.

Reply:
Yeah, sounds good!